### ISSUE TITLE
Premium theme not activated when signing up with Premium plan or above

### ISSUE THREAD

**Issue by seear** [Automattic/wp-calypso#21075]
#### Steps to reproduce
1. Starting at URL: (logged out of WordPress.com): https://wordpress.com/theme/photo-blog (or any other premium theme)
2. Select "Pick this design"
3. Complete the signup flow with at least a Premium plan

#### What I expected
Site should have the _photo-blog_ theme activated after checking out.

#### What happened instead
Site has a default theme activated.

#### Browser / OS version
n/a

#### Context / Source
Manual testing while investigating #20717

#### Notes
This happens because the Premium plan includes themes, so the theme is not added to the cart, but headstart does not allow switching to a premium theme (see `class-headstart-lifecycle#maybe_switch_to_theme` in Headstart repo).

**Comment by lancewillett** [comment]
I'd like to also fix this bad one: https://github.com/Automattic/wp-calypso/issues/14896

We shouldn't be charging customers for themes after they've upgraded.

**Comment by rachelmcr** [comment]
I tested and confirmed that this is still happening.

If I go to My Sites > Themes I can then activate the premium theme on my site at no additional charge, but if I go back to the URL I started at (https://wordpress.com/theme/photo-blog) I run into #14896 which is really bad.

**Comment by donpark** [comment]
Just started digging into this one.

**Comment by donpark** [comment]
Some ideas of best way to proceed:

1. To address #14896, change how we handle `/theme` to work like how `/customize` is handled: require site selection before showing themes. This will fix the problem of Premium Plan site owners having to pay for premium themes and avoid the need to refactoring `ThemeSheet` to support dynamically changing target site.

2. To address this issue of how to handle theme selection before site plan is chosen, I think we can either patch headstart to activate premium themes or, if that is blocked by some unknown conditions, schedule an auto-theme switch as last step in the headstart.

**Comment by donpark** [comment]
@seear "Pick this design" button currently does not allow me to create a new website a selected Premium theme. Is this issue still valid?

**Comment by ianstewart** [comment]
Adding this to the Design Selection program we're looking at for issue triage. Sounds fixed-ish? But we should test it. :)

**Comment by andres-blanco** [comment]
I confirm that this bug exists and it's reproducible.

A different theme needs to be used since the one cited above (photo-blog) does not exist anymore. You can use this: https://wordpress.com/theme/ovation

 I am searching for possible fixes, seems complex

**Comment by ianstewart** [comment]
> I am searching for possible fixes, seems complex

With the complexity and our priorities I'm going to close this one as wontfix. We can open it up again if there are user reports.
